parsl/tests/test_utils/test_execute_wait.py

```diff
@@ -0,0 +1,35 @@
+import pytest
+
+from parsl.utils import execute_wait
+
+
+@pytest.mark.local
+def test_env():
+    ''' Regression testing for issue #27
+    '''
+
+    rc, stdout, stderr = execute_wait("env", 1)
+
+    stdout = stdout.split('\n')
+    x = [s for s in stdout if s.startswith("PATH=")]
+    assert x, "PATH not found"
+
+    x = [s for s in stdout if s.startswith("HOME=")]
+    assert x, "HOME not found"
+
+
+@pytest.mark.local
+def test_large_output_2210():
+    """Regression test for #2210.
+    execute_wait was hanging if the specified command gave too
+    much output, due to a race condition between process exiting and
+    pipes filling up.
+    """
+
+    # this will output 128kb of stdout
+    execute_wait("yes | dd count=128 bs=1024", walltime=60)
+
+    # if this test fails, execute_wait should raise a timeout
+    # exception.
+
+    # The contents out the output is not verified by this test
```

Comment:
Similar question (and understanding of simply moving) as to the progeny of this test, though it's utility is more clear to me.

Meanwhile I'm unsure of the value of these particular numbers.  Why stop at "only" 128K?  If it's going to hang, perhaps we make _sure_ it does so we catch it, nominally with a much higher value.  1G would be overkill enough: `yes | dd count=1K bs=1M`

Similarly, if we're going to fail, this should fail super quick, why wait for a full minute?  5s should be beyond plenty of time, no?